services/src/main/scala/cromwell/services/metadata/impl/MetadataServiceActor.scala

```diff
@@ -28,7 +29,14 @@ object MetadataServiceActor {
 
 case class MetadataServiceActor(serviceConfig: Config, globalConfig: Config)
   extends Actor with ActorLogging with MetadataDatabaseAccess with SingletonServicesStore {
-
+  
+  override val supervisorStrategy = new OneForOneStrategy()(SupervisorStrategy.defaultStrategy.decider) {
+    override def logFailure(context: ActorContext, child: ActorRef, cause: Throwable, decision: Directive) = {
+      val childName = if (child == readActor) "Read" else "Write"
+      log.error(s"The $childName Metadata Actor died unexpectedly. Restarting it...", cause)
```

Comment:
I don't think I fully agree, all this actor does is taking messages, and calling a DB method. The DB method is encapsulated in a Future which makes error handling safe. Which means any other unexpected error will come from one of the messages in the data, so resuming will almost certainly cause the actor to perpetually fail. I don't see much room for transient errors not related to messages.